# Agriculture Crop Production
[https://data.world/agriculture/crop-production](https://data.world/agriculture/crop-production)

In [3]:
import pandas as pd
from glob import glob

In [4]:
## DATA
# area_code
# area
# item_code
# item
# element_code
# element
# unit
# y1961  ==>> données
# y1961f ==>> Flag pour cette donnée ( voir flags_df )
# y1962
# y1962f
# y1963,
# y1963f
# ....
# y2019

prod_df = pd.DataFrame()
for csv_file in glob('data/prod*.csv'):
    prod_df = pd.concat([prod_df, pd.read_csv(csv_file)])


## UNITS
# unit_name, description
units = pd.read_csv('data/units.csv')

## ITEMS
# domain_code, 
# domain, 
# item_code, 
# item, 
# description, 
# hs_code, 
# hs07_code, 
# hs12_code, 
# cpc_code
items_df = pd.read_csv('data/items.csv')

## FLAG, 
#  flag description (ex : Calculated data, Data not available, ...)
flags_df = pd.read_csv('data/flags.csv')

## ELEMENTS 
# domain_code, domain, element_code, element, unit
elements_df = pd.read_csv('data/elements.csv') 


In [21]:
prod_df.query('item_code == 515 and area == "Italy" and element == "Production"')

area_code   area  item_code    item  element_code     element    unit  \
4501        106  Italy        515  Apples          5510  Production  tonnes   

          y1961 y1961f      y1962  ...      y2015  y2015f      y2016  y2016f  \
4501  2167000.0    NaN  2182000.0  ...  2473608.0     NaN  2455616.0     NaN   

          y2017  y2017f      y2018  y2018f      y2019  y2019f  
4501  1921272.0     NaN  2466990.0     NaN  2303690.0     NaN  

[1 rows x 125 columns]

In [11]:
len(prod_df['item_code'].unique())

175

In [35]:
prod_df[['area']].value_counts()

area            
China, mainland     398
Mexico              373
Spain               361
Peru                330
Bulgaria            330
                   ... 
Tuvalu               27
Tokelau              27
China, Macao SAR     23
Marshall Islands     12
Faroe Islands         6
Length: 210, dtype: int64

# Dossiers
What's the difference ??
- un dossier **data**
- un dossier **original**

# Datasets
## Data **Production_crops_***

- **area_code**
  - 210 valeurs
- **area**
  - 210
- **item_code**
- **item**
- **element_code**
- **element**
- **unit**
- **y1961  ==>> données**
- **y1961f ==>> Flag pour cette donnée ( voir flags_df )**
- **y1962**
- **y1962f**
- **y1963,**
- **y1963f**
- **....**
- **y2019**

